CMakeLists.txt

```diff
@@ -80,6 +79,7 @@ if (${CMAKE_BUILD_TYPE} STREQUAL DEBUG)
 
     MESSAGE("cmake: DEBUG compilation")
     add_definitions(-DDEBUG -DDEBUG_$ENV{USER} -DDEBUG_${USER})
+    add_definitions(-fstack-check -fstack-protector)
```

Comment:
Suggestion for the comment:

"Note that stack protection is activated only in debug compilation. This is due that this directive makes the compiler introduce code at the begin/end of the functions, so has impact in performance (see <links where that is explained in more detail>)"